app/scripts/modules/kubernetes/src/rawResource/component/K8sResources.tsx

```diff
@@ -0,0 +1,111 @@
+import { Application, ApplicationDataSource } from '@spinnaker/core';
+import React from 'react';
+import { FiltersPubSub } from '../controller/FiltersPubSub';
+import { KUBERNETS_RAW_RESOURCE_DATA_SOURCE_KEY } from '../rawResource.dataSource';
+import { RawResource } from './group/RawResource';
+import { RawResourceGroups } from './group/RawResourceGroups';
+import { IK8sResourcesFiltersState } from './K8sResourcesFilters';
+
+export interface IK8sResourcesProps {
+  app: Application;
+}
+
+interface IK8sResourcesState {
+  groupBy: string;
+  filters: IK8sResourcesFiltersState;
+  rawResources: IApiKubernetesResource[];
+}
+
+export class K8sResources extends React.Component<IK8sResourcesProps, IK8sResourcesState> {
+  private dataSource: ApplicationDataSource<IApiKubernetesResource[]>;
+  private filterPubSub: FiltersPubSub = FiltersPubSub.getInstance(this.props.app.name);
+
+  constructor(props: IK8sResourcesProps) {
+    super(props);
+    this.dataSource = this.props.app.getDataSource(KUBERNETS_RAW_RESOURCE_DATA_SOURCE_KEY);
+    this.state = {
+      groupBy: 'none',
+      filters: null,
+      rawResources: [],
+    };
+
+    this.filterPubSub.subscribe(this.onFilterChange.bind(this));
+  }
+
+  public onFilterChange(message: IK8sResourcesFiltersState) {
+    this.setState({ ...this.state, filters: message });
+  }
+
+  public async componentDidMount() {
+    await this.dataSource.ready();
+
+    this.setState({
+      ...this.state,
+      groupBy: this.state.groupBy,
+      rawResources: await this.dataSource.data.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1)),
+    });
+  }
+
+  public onGroupByChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({
+      ...this.state,
+      groupBy: event.target.value,
+    });
+  }
+
+  public render() {
+    return (
+      <div style={{ width: '100%' }}>
+        <form className="form-inline" style={{ marginBottom: '5px' }}>
+          <div className="form-group">
```

Comment:
Can you use the `FormikFormField` (preferred) or `FormField` (not preferred, but better than raw html) and `SelectInput` instead?